modules/apps/object/object-web/src/test/java/com/liferay/object/web/internal/notifications/ObjectUserNotificationsHandlerTest.java

```diff
@@ -0,0 +1,217 @@
+ */
+
+package com.liferay.object.web.internal.notifications;
+
+import com.liferay.asset.display.page.portlet.AssetDisplayPageFriendlyURLProvider;
+import com.liferay.info.item.InfoItemReference;
+import com.liferay.object.model.ObjectDefinition;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.UserNotificationEvent;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+
+import javax.portlet.PortletURL;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+/**
+ * @author Thalles Montenegro
+ */
+public class ObjectUserNotificationsHandlerTest {
+
+	@ClassRule
+	public static LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		_objectUserNotificationsHandler = new ObjectUserNotificationsHandler(
+			_assetDisplayPageFriendlyURLProvider, _objectDefinition);
+
+		_portletURLStep = new PortletURLBuilder.PortletURLStep(
+			Mockito.mock(PortletURL.class));
+
+		_portletURL = Mockito.mock(PortletURL.class);
+	}
+
+	@Test
+	public void testGetLinkWhenNotificationHasFriendlyURL() throws Exception {
+		Mockito.when(
+			_userNotificationEvent.getPayload()
+		).thenReturn(
+			""
+		);
+
+		Mockito.when(
+			_serviceContext.getThemeDisplay()
+		).thenReturn(
+			Mockito.mock(ThemeDisplay.class)
+		);
+
+		Mockito.when(
+			_assetDisplayPageFriendlyURLProvider.getFriendlyURL(
+				Mockito.any(InfoItemReference.class),
+				Mockito.any(ThemeDisplay.class))
+		).thenReturn(
+			"friendlyURL"
+		);
+
+		Assert.assertEquals(
+			"friendlyURL",
+			_objectUserNotificationsHandler.getLink(
+				_userNotificationEvent, _serviceContext));
+	}
+
+	@Test
+	public void testGetLinkWhenNotificationIsCreatedAfterGuestUserAddEntryThatExceedsLimits()
+		throws Exception {
+
+		Mockito.when(
+			_serviceContext.getThemeDisplay()
+		).thenReturn(
+			null
+		);
+
+		Mockito.when(
+			_serviceContext.getRequest()
+		).thenReturn(
+			Mockito.mock(HttpServletRequest.class)
+		);
+
+		Mockito.when(
+			_serviceContext.getCurrentURL()
+		).thenReturn(
+			RandomTestUtil.randomString()
+		);
+
+		Mockito.when(
+			_userNotificationEvent.getPayload()
+		).thenReturn(
+			"{ \"exceedsObjectEntryLimit\": true }"
+		);
+
+		try (MockedStatic<RequestBackedPortletURLFactoryUtil>
+				factoryUtilMockedStatic = Mockito.mockStatic(
```

Comment:
Rename `factoryUtilMockedStatic` to `requestBackedPortletURLFactoryUtilMockedStatic`

Thanks!